rest/rest_client.go

```diff
@@ -121,7 +121,7 @@ func (c *clientImpl) retry(endpoint *CompiledEndpoint, rqBody any, rsBody any, t
 	if err != nil {
 		return fmt.Errorf("error locking bucket in rest client: %w", err)
 	}
-	rq = rq.WithContext(config.Ctx)
+	config.Request = config.Request.WithContext(config.Ctx)
```

Comment:
I think it would make more sense to override `rq` here to prevent future misshandling

```suggestion
	rq = config.Request.WithContext(config.Ctx)
```

in line 133 this would require the usage of `rq` too
```diff
- rs, err := c.HTTPClient().Do(config.Request)
+ rs, err := c.HTTPClient().Do(rq)
```

Reply:
sounds dangerous - then any customization made via config might be lost? even if it's a pointer - in the future this kind of bug might happen again because someone will replace the ptr in the config

Reply:
Nope it won't be lost since we would do
```go
rq = config.Request.WithContext(config.Ctx)
```

if we did
```go
rq = rq.WithContext(config.Ctx)
```
This would apply, but my suggestion is the first

Reply:
ahhh I see! one second